test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs

```diff
@@ -860,7 +861,7 @@ private static RouteContext CreateRouteContext(string httpMethod)
             var routeData = new RouteData();
             routeData.Routers.Add(new Mock<IRouter>(MockBehavior.Strict).Object);
 
-            var httpContext = new Mock<HttpContext>(MockBehavior.Strict);
+            var httpContext = new Mock<HttpContext>(MockBehavior.Loose);
```

Comment:
We are going to make it strict, and pass services, as well as remove setting services on HttpContext from the test above